test/e2e/query_test.go

```diff
@@ -849,6 +850,344 @@ func TestSidecarStorePushdown(t *testing.T) {
 	})
 }
 
+type seriesWithLabels struct {
+	intLabels labels.Labels
+	extLabels labels.Labels
+}
+
+func TestQueryStoreDedup(t *testing.T) {
```

Comment:
Can we  add `// Regression test for <issue id>

Reply:
Or actually better do it on specific test case (: 

And you did it already. Thanks!